javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java

```diff
@@ -341,13 +318,19 @@ && shouldGeneratePagingMethods()) {
                     if (proxyMethod.isSync()) {
                         continue;
                     }
+
                     JavaVisibility simpleAsyncMethodVisibility =
-                        methodVisibility(ClientMethodType.SimpleAsyncRestResponse, false, isProtocolMethod);
+                        methodVisibility(ClientMethodType.SimpleAsyncRestResponse, defaultOverloadType, false, isProtocolMethod);
                     JavaVisibility simpleAsyncMethodVisibilityWithContext =
-                        methodVisibility(ClientMethodType.SimpleAsyncRestResponse, true, isProtocolMethod);
+                        methodVisibility(ClientMethodType.SimpleAsyncRestResponse, defaultOverloadType, true, isProtocolMethod);
+                    // for vanilla and fluent, the SimpleAsyncRestResponse is VISIBLE, so that they can be used for possible customization on LRO
+
+                    // there is ambiguity of RestResponse from simple API and from LRO API
```

Comment:
nit, maybe we can put this override into `methodVisibility()`? There's also `settings.isDataPlaneClient()` logic.